components/TheOverlayComponent.vue

```diff
@@ -0,0 +1,212 @@
+<template>
+  <no-ssr>
+    <div class="the-overlay-component">
+
+      <div v-if="shouldShowWeb3Dialog">
+        <div v-if="checkIsMobileClient()">
+          <!-- Trust Dialog start (show when mobile) -->
+          <v-dialog
+            v-model="isDialogOpen"
+            max-width="290"
+            @keydown.esc="isDialogOpen = false"
+          >
+            <v-card>
+              <v-card-title class="headline">
+                Wallet App required
+              </v-card-title>
+            </v-card>
+            <v-btn dark flat @click.native="isDialogOpen = false">Close</v-btn>
+          </v-dialog>
+          <!-- Trust Dialog end -->
+        </div>
+        <div v-else>
+          <!-- Chrome Dialog start (ask when user use browser like Safari) -->
+          <v-dialog
+            v-model="isDialogOpen"
+            max-width="290"
+            v-if="shouldShowChromeDialog"
+            @keydown.esc="isDialogOpen = false"
+          >
+            <v-card>
+              <v-card-title class="headline">
+                Chrome required
+              </v-card-title>
+            </v-card>
+            <v-btn dark flat @click.native="isDialogOpen = false">Close</v-btn>
+          </v-dialog>
+          <!-- Chrome Dialog end -->
+
+          <!-- Metamask related dialog start -->
+          <v-dialog
```

Comment:
Please refactor this dialog to another single file for better readability, since we want to place all global dialogs in this component